src/Stratis.Bitcoin.Features.BlockStore/BlockStoreQueue.cs

```diff
@@ -181,17 +184,21 @@ public async Task<Block> GetBlockAsync(uint256 blockHash)
         {
             this.logger.LogTrace("({0}:'{1}')", nameof(blockHash), blockHash);
 
-            Block block = null;
-
             lock (this.getBlockLock)
             {
-                block = this.batch.FirstOrDefault(x => x.ChainedHeader.HashBlock == blockHash)?.Block;
+                if (this.blocksBeingPersisted.TryGetValue(blockHash, out ChainedHeaderBlock chainedHeaderBlock))
+                {
+                    this.logger.LogTrace("(-)[FOUND_IN_DICTIONARY]");
+                    return chainedHeaderBlock.Block;
+                }
             }
 
+            Block block = await this.blockRepository.GetBlockAsync(blockHash).ConfigureAwait(false);
+
             if (block == null)
-                block = await this.blockRepository.GetBlockAsync(blockHash).ConfigureAwait(false);
+                this.logger.LogTrace("Block was not found in the repository.");
```

Comment:
those 4 lines can be replaced with 
`this.logger.LogTrace("Block was{0} found in the repository.", (block == null)? " not" : "");`